pkg/observer/observer.go

```diff
@@ -226,7 +238,12 @@ func (k *Observer) RunEvents(stopCtx context.Context, ready func()) error {
 				}
 			} else {
 				if len(record.RawSample) > 0 {
-					k.receiveEvent(record.RawSample)
+					select {
+					case eventsQueue <- &record:
+					default:
+						// eventsQueue channel is full, drop the event
```

Comment:
~I think we need to count these lost events somehow.~
Nevermind, it's in the second patch.